plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

```diff
@@ -498,101 +518,67 @@ private static ColumnMapping varcharColumnMapping(int varcharLength, Optional<Ca
+    private static ObjectWriteFunction longTimestampWriteFunction(int precision)
     {
         return ObjectWriteFunction.of(
                 LongTimestamp.class,
                 (statement, index, value) -> statement.setString(index, SNOWFLAKE_TIMESTAMP_FORMATTER.format(StandardColumnMappings.fromLongTrinoTimestamp(value, precision))));
     }
 
-    private static WriteMapping snowFlakeTimestampWithTZWriter(Type type)
+    private static WriteMapping snowflakeTimestampWithTimeZoneWriteMapping(TimestampWithTimeZoneType timestampWithTimeZoneType)
     {
-        TimestampWithTimeZoneType timeTZType = (TimestampWithTimeZoneType) type;
-
-        checkArgument(timeTZType.getPrecision() <= MAX_SUPPORTED_TEMPORAL_PRECISION, "Max Snowflake precision is is " + MAX_SUPPORTED_TEMPORAL_PRECISION);
-        if (timeTZType.isShort()) {
-            return WriteMapping.longMapping(format("timestamp_tz(%d)", timeTZType.getPrecision()), timestampWithTimezoneWriteFunction());
+        checkArgument(timestampWithTimeZoneType.getPrecision() <= MAX_SUPPORTED_TEMPORAL_PRECISION, "Max Snowflake precision is is " + MAX_SUPPORTED_TEMPORAL_PRECISION);
+        if (timestampWithTimeZoneType.isShort()) {
+            return WriteMapping.longMapping(format("timestamp_tz(%d)", timestampWithTimeZoneType.getPrecision()), shortTimestampWithTimeZoneWriteFunction());
         }
-        return WriteMapping.objectMapping(format("timestamp_tz(%d)", timeTZType.getPrecision()), longTimestampWithTzWriteFunction());
+        return WriteMapping.objectMapping(format("timestamp_tz(%d)", timestampWithTimeZoneType.getPrecision()), longTimestampWithTimeZoneWriteFunction());
     }
 
-    private static LongWriteFunction timestampWithTimezoneWriteFunction()
+    private static LongWriteFunction shortTimestampWithTimeZoneWriteFunction()
     {
         return (statement, index, encodedTimeWithZone) -> {
             Instant instant = Instant.ofEpochMilli(DateTimeEncoding.unpackMillisUtc(encodedTimeWithZone));
             ZoneId zone = ZoneId.of(DateTimeEncoding.unpackZoneKey(encodedTimeWithZone).getId());
             statement.setString(index, SNOWFLAKE_DATETIME_FORMATTER.format(instant.atZone(zone)));
         };
     }
+
+    private static ObjectWriteFunction longTimestampWithTimeZoneWriteFunction()
+    {
+        return ObjectWriteFunction.of(LongTimestampWithTimeZone.class, (statement, index, value) -> {
+            long epoMilli = value.getEpochMillis();
+            long epoSeconds = Math.floorDiv(epoMilli, Timestamps.MILLISECONDS_PER_SECOND);
+            long adjNano = (long) Math.floorMod(epoMilli, Timestamps.MILLISECONDS_PER_SECOND) * Timestamps.NANOSECONDS_PER_MILLISECOND + value.getPicosOfMilli() / Timestamps.PICOSECONDS_PER_NANOSECOND;
+            ZoneId zone = TimeZoneKey.getTimeZoneKey(value.getTimeZoneKey()).getZoneId();
+            Instant timeI = Instant.ofEpochSecond(epoSeconds, adjNano);
```

Comment:
Please avoid abbreviations: https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-abbreviations. I know it comes from the existing code, but git diff shows as a new code.